docs/configuration.rst

```diff
@@ -79,6 +86,41 @@ Options
 
 All configuration options starts with the prefix ``needs_`` for **Sphinx-Needs**.
 
+.. _needs_from_toml:
+
+needs_from_toml
+~~~~~~~~~~~~~~~
+
+.. versionadded:: 4.1.0
+
+This configuration takes the (relative) path to a `toml file <https://toml.io>`__ which contains some or all of the needs configuration (configuration in the toml will override that in the :file:`conf.py`).
+
+.. code-block:: python
+
+   needs_from_toml = "ubproject.toml"
+
+Configuration in the toml can contain any of the options below, under a ``[needs]`` section,
+but with the ``needs_`` prefix removed.
+For example:
+
+.. code-block:: toml
+
+   [needs]
+   id_required = true
+   id_length = 3
+   types = [
+      {directive="req", title="Requirement", prefix="R_", color="#BFD8D2", style="node"},
+      {directive="spec", title="Specification", prefix="S_", color="#FEDCD2", style="node"},
+   ]
+
+To specify a different section name to read from in the toml file, use the ``needs_from_toml_section`` option:
+
+.. code-block:: python
+
+   needs_from_toml_section = ["tool", "needs"]
```

Comment:
Only understood this after reading the code.
Is section the right term? It's a table in toml. 

Can I use tool.needs in the list?

Would also change to plural 'sections' (or tables).